enhancements/storage/csi-driver-install.md

```diff
@@ -151,40 +153,76 @@ When a CSI driver is moved from optional to a mandatory one, existing installati
 
 1. CVO installs cluster-storage-operator.
 2. cluster-storage-operator detects cloudprovider on which cluster is running(lets say EBS).
-3. cluster-storage-operator searches for existing subscription to an operator with same name in all namespaces and if found it deletes and re-creates the subscription in pre-defined namespace.
-4. At this point if there are any changes in configuration CR between previously installed versions vs version shipped with new operator, it must be reconciled.
+3. cluster-storage-operator searches for existing subscription to an operator with same name in all namespaces and if found it deletes it and re-creates the subscription in pre-defined namespace.
+4. At this point if there are any changes in configuration CR between previously installed versions vs version shipped with new operator, they are reconciled.
+
+### Release
+
+Since operators managed by OLM are not managed ART pipeline, there are some consequences.
+
+1. CSI sidecar images, used by the operator, are part of OCP extras errata.
+   Therefore at OCP 4.x release (or shortly after), we must rebuild the operator bundle image with sidecar SHAs that were just released by OCP and respin the operator errata.
 
+   The operator will **not** be available at OCP GA date, it will be released later (few days?).
+      * When the operator is optional, it's not an issue, cusomers can use in-tree volume plugins and experiment with CSI few days later.
+      * When the operator is installed by default and an older version of the operator is available, it's not an issue.
+        OCP installation will use the old version of the operator until the new version is available.
+      * When the operator is installed by default and an older version of the operator is not available, **OCP cluster cannot be installed** until the operator errata is released.
+          * The operator and OCP erratas should be synchronized and released at the same time.
+            The operator bugs blocks OCP from release.
+            This applies only to the first release of the operator, when there is no older version to use.
+            It may not be an issue if we follow the timeline outlined below, but it's a very tight timeline!
 
-### Open questions for OLM team:
+2. There is no automation to rebuild operator bundle images when operand or sidecar image change.
+   In addition we must sync dist-gits manually and create / respin erratas manually too:
+   * During development & testing, we will update the operator / operand / sidecar SHAs only few times (say once per sprint).
+   * After release, we won't release any z-streams of the operators, CSI drivers and/or SHAs of new sidecars unless absolutelly necessary (CVEs, urgent bugs).
+
+3. Since there is no CI for images that we build ourselves (without ART), all our downstream images are tested only by QA.
```

Comment:
Upstream CI testing and downstream CVP testing should still run. This is one area where ART doesn't add anything.

Reply:
@jupierce do you have a link to CVP? Never heard it of it.

Reply:
https://docs.engineering.redhat.com/pages/viewpage.action?spaceKey=CVP&title=Container+Verification+Pipeline+E2E+Documentation#ContainerVerificationPipelineE2EDocumentation-Operatormetadatatestingpipeline
There is a TOC entry for operators. Eventually it is supposed to include custom functional tests. Presently, it verifies metadata and the ability to install.

Reply:
I am mostly worried about manual SHA replacement now. I know that it may get better soon-ish, still, until the automation is in place, I really don't know if the bundle image is correct.